docs/pallas/tpu/distributed.md

```diff
@@ -0,0 +1,1313 @@
TI1IDAgLTEuNjU2MjUgMC40MDYyNXEtMC42ODc1IDAuMzkwNjI1IC0xLjEyNSAxLjE4NzVsNS41NDY4NzUgMHptOC41NTQ5MzIgLTEuMzQzNzVsMCA0LjUzMTI1NzZsMi42NzE4NzUgMHEwLjY3MTg3NSAwIDAuOTUzMTI1IDAuMjY1NjI1cTAuMjgxMjUgMC4yNjU2MjUgMC4yODEyNSAwLjY4NzVxMCAwLjQwNjI1IC0wLjI4MTI1IDAuNjcxODc1cS0wLjI4MTI1IDAuMjUgLTAuOTUzMTI1IDAuMjVsLTUuNjQwNjI1IDBxLTAuNjcxODc1IDAgLTAuOTY4NzUgLTAuMjVxLTAuMjgxMjUgLTAuMjY1NjI1IC0wLjI4MTI1IC0wLjY4NzVxMCAtMC40MDYyNSAwLjI4MTI1IC0wLjY3MTg3NXEwLjI5Njg3NSAtMC4yNjU2MjUgMC45Njg3NSAtMC4yNjU2MjVsMS4xMDkzNzUgMGwwIC00LjUzMTI1NzZsLTAuODkwNjI1IDBxLTAuNjcxODc1IDAgLTAuOTY4NzUgLTAuMjVxLTAuMjgxMjUgLTAuMjY1NjI1IC0wLjI4MTI1IC0wLjY4NzVxMCAtMC40MDYyNSAwLjI4MTI1IC0wLjY1NjI1cTAuMjk2ODc1IC0wLjI2NTYyNSAwLjk2ODc1IC0wLjI2NTYyNWwwLjg5MDYyNSAwbDAgLTAuNzE4NzVxMCAtMS4yMTg3NSAwLjkwNjI1IC0yLjAxNTYyNXEwLjkyMTg3NSAtMC44MTI1IDIuNjU2MjUgLTAuODEyNXEwLjc4MTI1IDAgMS43OTY4NzUgMC4xNDA2MjVxMS4wMTU2MjUgMC4xNDA2MjUgMS4yOTY4NzUgMC40MDYyNXEwLjI5Njg3NSAwLjI1IDAuMjk2ODc1IDAuNjQwNjI1cTAgMC40Mzc1IC0wLjI2NTYyNSAwLjcwMzEyNXEtMC4yNSAwLjI2NTYyNSAtMC42MjUgMC4yNjU2MjVxLTAuMTcxODc1IDAgLTAuNTE1NjI1IC0wLjA2MjVxLTEuMTU2MjUgLTAuMjM0Mzc1IC0yLjA0Njg3NSAtMC4yMzQzNzVxLTAuOTM3NSAwIC0xLjI5Njg3NSAwLjI4MTI1cS0wLjM0Mzc1IDAuMjgxMjUgLTAuMzQzNzUgMC42ODc1bDAgMC43MTg3NWwyLjg5MDYyNSAwcTAuNjcxODc1IDAgMC45NTMxMjUgMC4yNjU2MjVxMC4yODEyNSAwLjI1IDAuMjgxMjUgMC42NzE4NzVxMCAwLjQwNjI1IC0wLjI4MTI1IDAuNjcxODc1cS0wLjI4MTI1IDAuMjUgLTAuOTUzMTI1IDAuMjVsLTIuODkwNjI1IDB6IiBmaWxsLXJ1bGU9Im5vbnplcm8iLz48cGF0aCBmaWxsPSIjZDlkOWQ5IiBkPSJtNzQ2LjIzODcgMTY0LjUxNjhsMTkyLjEyNTk4IDBsMCA0Mi4wMTU3NDdsLTE5Mi4xMjU5OCAweiIgZmlsbC1ydWxlPSJldmVub2RkIi8+PHBhdGggZmlsbD0iIzAwMDAwMCIgZD0ibTgxMi40NjE4NSAxOTEuNDM2OGwwIC0xLjIxODc1cS0wLjkwNjI1IDEuNDM3NSAtMi43MDMxMjUgMS40Mzc1cS0xLjE1NjI1IDAgLTIuMTI1IC0wLjY0MDYyNXEtMC45Njg3NSAtMC42NDA2MjUgLTEuNSAtMS43ODEyNXEtMC41MzEyNSAtMS4xNDA2MjUgLTAuNTMxMjUgLTIuNjI1cTAgLTEuNDUzMTI1IDAuNDg0Mzc1IC0yLjYyNXEwLjQ4NDM3NSAtMS4xODc1IDEuNDM3NSAtMS44MTI1cTAuOTY4NzUgLTAuNjI1IDIuMTcxODc1IC0wLjYyNXEwLjg3NSAwIDEuNTQ2ODc1IDAuMzc1cTAuNjg3NSAwLjM1OTM3NSAxLjEwOTM3NSAwLjk1MzEyNWwwIC00Ljc5Njg3NWwxLjY0MDYyNSAwbDAgMTMuMzU5Mzc1bC0xLjUzMTI1IDB6bS01LjE3MTg3NSAtNC44MjgxMjVxMCAxLjg1OTM3NSAwLjc4MTI1IDIuNzgxMjVxMC43ODEyNSAwLjkyMTg3NSAxLjg0Mzc1IDAuOTIxODc1cTEuMDc4MTI1IDAgMS44MjgxMjUgLTAuODc1cTAuNzUgLTAuODkwNjI1IDAuNzUgLTIuNjg3NXEwIC0xLjk4NDM3NSAtMC43NjU2MjUgLTIuOTA2MjVxLTAuNzY1NjI1IC0wLjkzNzUgLTEuODkwNjI1IC0wLjkzNzVxLTEuMDc4MTI1IDAgLTEuODEyNSAwLjg5MDYyNXEtMC43MzQzNzUgMC44OTA2MjUgLTAuNzM0Mzc1IDIuODEyNXptMTUuOTA2OTgyIDEuNzE4NzVsMS42ODc1IDAuMjAzMTI1cS0wLjQwNjI1IDEuNDg0Mzc1IC0xLjQ4NDM3NSAyLjMxMjVxLTEuMDc4MTI1IDAuODEyNSAtMi43NjU2MjUgMC44MTI1cS0yLjEyNSAwIC0zLjM3NSAtMS4yOTY4NzVxLTEuMjM0Mzc1IC0xLjMxMjUgLTEuMjM0Mzc1IC0zLjY3MTg3NXEwIC0yLjQ1MzEyNSAxLjI1IC0zLjc5Njg3NXExLjI2NTYyNSAtMS4zNDM3NSAzLjI2NTYyNSAtMS4zNDM3NXExLjkzNzUgMCAzLjE1NjI1IDEuMzI4MTI1cTEuMjM0Mzc1IDEuMzEyNSAxLjIzNDM3NSAzLjcwMzEyNXEwIDAuMTU2MjUgMCAwLjQzNzVsLTcuMjE4NzUgMHEwLjA5Mzc1IDEuNTkzNzUgMC45MDYyNSAyLjQ1MzEyNXEwLjgxMjUgMC44NDM3NSAyLjAxNTYyNSAwLjg0Mzc1cTAuOTA2MjUgMCAxLjU0Njg3NSAtMC40Njg3NXEwLjY0MDYyNSAtMC40ODQzN
```

Comment:
"Therefore, we can operate along one dimension using a ring communication pattern, and use the other dimensions for batching."

I'd say something more like:

"When writing collectives, we often only need to think about 1D slices of our torus at a time because the different dimensions of the torus are reserved for different types of parallelism (data vs. model, for example)."

Reply:
Done.